src/Cli/dotnet/commands/dotnet-workload/install/MsiInstallerBase.cs

```diff
@@ -525,23 +562,23 @@ protected void RemoveInstallStateFile(SdkFeatureBand sdkFeatureBand)
         /// </summary>
         /// <param name="sdkFeatureBand">The path of the isntall state file to write.</param>
         /// <param name="jsonLines">The contents of the JSON file, formatted as a single line.</param>
-        protected void WriteInstallStateFile(SdkFeatureBand sdkFeatureBand, IEnumerable<string> jsonLines)
+        protected void WriteInstallStateFile(string dotnetDir, SdkFeatureBand sdkFeatureBand, Dictionary<string, string> jsonLines)
```

Comment:
Naming: Since this method is no longer being passed all of the data that goes into the install state file, I think it might be clearer if it (and other similar methods) were renamed to something like `SaveInstallStateManifestVersions`.  The `jsonLines` parameter could be renamed to `manifestVersions`.